src/components/EmailNotifications/EmailNotificationsProvider.js

```diff
@@ -0,0 +1,142 @@
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react'
+import PropTypes from 'prop-types'
+import { useWallet } from '../../providers/Wallet'
+import {
+  getJurorEmail,
+  getSubscriptionDetails,
+  subscribeExistingEmail,
+  subscribeToNotifications,
+} from '../../services/servicesRequests'
+
+const EmailNotificationsContext = React.createContext()
+
+function EmailNotificationsProvider({ children }) {
+  const { account } = useWallet()
+
+  const asyncCancelled = useRef(false)
+  const [email, setEmail] = useState(null)
+  const [needsSignature, setNeedsSignature] = useState(false)
+  const [subscriptionDetails, setSubscriptionDetails] = useState({
+    error: null,
+    fetching: false,
+  })
+
+  const handleOnSubscribe = useCallback(
+    async email => {
+      const response = await subscribeToNotifications(account, email)
+
+      if (response.error && !response.needsSignature) {
+        return response.error
```

Comment:
Thinking about this, we may want to throw the error instead (and perhaps all the requests should be throwing those errors, similar to https://github.com/aragon/aragon/blob/master/src/components/GlobalPreferences/Notifications/notification-service-api.js#L60).